ingest/jobs/Ingest_AzureSql_Example/tests/end_to_end/features/fixtures.py

```diff
@@ -0,0 +1,30 @@
+from azure.identity import DefaultAzureCredential
+from azure.storage.filedatalake import DataLakeServiceClient
+from behave import fixture
+from constants import ADLS_URL, RAW_CONTAINER_NAME, AUTOMATED_TEST_OUTPUT_DIRECTORY_PREFIX
+from utils.azure.adls import filter_directory_paths_adls, delete_directories_adls
+
+@fixture
+def azure_adls_clean_up(context, ingest_directory_name: str):
+    credential = DefaultAzureCredential()
+    adls_client = DataLakeServiceClient(account_url=ADLS_URL, credential=credential)
+    print('BEFORE SCENARIO. DELETING ANY AUTOMATED TEST OUTPUT DATA')
+    automated_test_output_directory_paths = filter_directory_paths_adls(adls_client,
+                                                                        RAW_CONTAINER_NAME,
+                                                                        ingest_directory_name,
+                                                                        AUTOMATED_TEST_OUTPUT_DIRECTORY_PREFIX)
+
+    if automated_test_output_directory_paths:
+        delete_directories_adls(adls_client, RAW_CONTAINER_NAME, automated_test_output_directory_paths)
```

Comment:
This block is duplicated (here and in the other tests). Apart from `if` being redundant, maybe extract it to a separate function?

Reply:
Yeah I can extract this

Reply:
Actually on second thoughts. I don't think it makes sense. It's literally just combining two functions together, both of which already have a clear purpose. I'd rather keep these separate for transparency.
@motylp

Reply:
Ok, my preference would be to extract, but in such cases I'm leaving the final decision to the author :) Resolving.